File: services/ticket-service/internal/domain/ticket.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Ticket struct {
	ID         uuid.UUID
	ProjectID  uuid.UUID
	Identifier string

	Title       string
	Description string

	Status    TicketStatus
	Priority  TicketPriority
	Tags      []string
	CreatedAt time.Time
	UpdatedAt time.Time
}

type TicketPriority string

const (
	PriorityCritical TicketPriority = "critical"
	PriorityHigh     TicketPriority = "high"
	PriorityMedium   TicketPriority = "medium"
	PriorityLow      TicketPriority = "low"
)

type TicketStatus string

const (
	StatusBacklog    TicketStatus = "backlog"
	StatusInProgress TicketStatus = "in_progress"
	StatusInReview   TicketStatus = "in_review"
	StatusDone       TicketStatus = "done"
)

File: web.legacy/src/lib/types/ticket.ts
export type TicketPriority = 'low' | 'medium' | 'high' | 'critical';
export type TicketStatus = 'backlog' | 'in_progress' | 'in_review' | 'done';

export interface Ticket {
	ID: string;
	ProjectID: string;
	Identifier: string;
	Title: string;
	Description: string;
	Status: TicketStatus;
	Priority: TicketPriority;
	CreatedBy: string;
	Assignee?: string | null;
	DueDate?: string | null;
	CreatedAt: string;
	UpdatedAt: string;
}

export interface CreateTicketPayload {
	project_id: string;
	identifier: string;
	title: string;
	description: string;
	priority: TicketPriority;
	created_by: string;
}
